mindsdb/integrations/handlers/neuralforecast_handler/neuralforecast_handler.py

```diff
@@ -0,0 +1,99 @@
+import os
+import string
+import random
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.integrations.handlers.statsforecast_handler.statsforecast_handler import (
+    infer_frequency,
+    transform_to_nixtla_df,
+    get_results_from_nixtla_df,
+)
+from neuralforecast import NeuralForecast
+from neuralforecast.models import NHITS
+from neuralforecast.auto import AutoNHITS
+from ray.tune.search.hyperopt import HyperOptSearch
+
+DEFAULT_FREQUENCY = "D"
+DEFAULT_MODEL_NAME = "NHITS"
+DEFAULT_TRIALS = 20
+MIN_TRIALS_FOR_AUTOTUNE = 3
+
+
+def choose_model(num_trials, horizon, window, exog_vars=[], threshold=MIN_TRIALS_FOR_AUTOTUNE):
+    """Chooses model based on the number of trials allowed by the user.
+
+    A lower args for training time reduces the number of trial models we can
+    search through. Below a certain threshold, we switch to the default
+    NHITS implementation instead of using AutoML to search for the best config.
+    """
+    if num_trials >= threshold:
+        model = AutoNHITS(horizon, gpus=0, num_samples=num_trials, search_alg=HyperOptSearch())
+    else:  # faster implementation without auto parameter tuning
+        model = NHITS(horizon, window, hist_exog_list=exog_vars)
+    return model
+
+
+class NeuralForecastHandler(BaseMLEngine):
+    """Integration with the Nixtla NeuralForecast library for
+    time series forecasting with neural networks.
+    """
+
+    name = "neuralforecast"
+
+    def create(self, target, df, args={}):
+        """Create the NeuralForecast Handler.
+
+        Requires specifying the target column to predict and time series arguments for
+        prediction horizon, time column (order by) and grouping column(s).
+
+        Saves model params to desk, which are called later in the predict() method.
+        """
+        with open("args.txt", "w+") as f:
```

Comment:
Is this actually being used? If so, please use `tempfile.mkdtemp` as base directory instead.

Reply:
It's not being used, you caught some dirty stuff I did when debugging 😬 Nice spot.